python/src/com/jetbrains/python/codeInsight/imports/PythonImportUtils.java

```diff
@@ -234,20 +237,31 @@ private static boolean isPossibleModuleReference(PyElement node) {
 
   private static Collection<PsiElement> findImportableModules(PsiFile targetFile, String reftext, Project project, GlobalSearchScope scope) {
     List<PsiElement> result = new ArrayList<>();
-    PsiFile[] files = FilenameIndex.getFilesByName(project, reftext + ".py", scope);
-    for (PsiFile file : files) {
-      if (isImportableModule(targetFile, file)) {
-        result.add(file);
+    //add packages
+    final PsiManager psiMgr = PsiManager.getInstance(project);
+    FileBasedIndex.getInstance().processValues(FilenameIndex.NAME, reftext, null, (file, value) -> {
```

Comment:
`processFilesByName(name, includeDirs = true, ...)` searches only for directories that match `name`, so extra `VirtualFile -> PsiDirectory` happen only for those dirs that match the name but don't contain `__init__.py`. Do you notice any considerable slowdown if you switch from `processValues` to `processFilesByName`?